src/helpers/spaces.ts

```diff
@@ -145,6 +145,57 @@ async function getVotes() {
   return await db.queryAsync(query);
 }
 
+export async function getCombinedMembersAndVoters(
+  spaceId,
+  cursor = null,
+  pageSize
+) {
+  const params = [spaceId, spaceId, spaceId, spaceId];
+  const query = `
+    SELECT address
+    FROM (
+      (SELECT DISTINCT JSON_UNQUOTE(JSON_EXTRACT(settings, '$.admins[*]')) AS address 
+       FROM spaces 
+       WHERE id = ?)
+      UNION
+      (SELECT DISTINCT JSON_UNQUOTE(JSON_EXTRACT(settings, '$.moderators[*]')) AS address 
+       FROM spaces 
+       WHERE id = ?)
+      UNION
+      (SELECT DISTINCT JSON_UNQUOTE(JSON_EXTRACT(settings, '$.members[*]')) AS address 
+       FROM spaces 
+       WHERE id = ?)
+      UNION
+      (SELECT DISTINCT voter AS address
+       FROM votes
+       WHERE space = ?)
+    ) AS combined_addresses
+    WHERE (? IS NULL OR address > ?)
+    ORDER BY address
+    LIMIT ?
+  `;
```

Comment:
This is actually taking 22 seconds for spaces like stgdao.eth

Reply:
Noted. 22 secs for a DAO with 912K members